tests/moe/test_unified_moe_fp8.py

```
@@ -765,6 +821,19 @@ def _assert_per_tensor_fp8_close(out: torch.Tensor, ref: torch.Tensor) -> None:
     check_accuracy(out.float(), ref.float(), atol=0.05, rtol=0.3, percent=0.99)
 
 
+def _assert_per_tensor_fp8_discriminates(out: torch.Tensor, ref: torch.Tensor) -> None:
+    """Negative control for the percentage-based tolerance above.
+
+    ``check_accuracy`` passes when 99% of elements are within a generous
+    rtol=0.3, so a pass alone does not prove the comparison could detect a wrong
+    activation formula. Confirm a perturbed reference is actually rejected.
+    """
+    with pytest.raises(Exception, match="[Mm]ismatch"):
```

Ruff (0.16.1)

[warning] Pattern passed to `match=` contains metacharacters but is neither escaped nor raw

(RUF043)